nautobot/core/models/tree_queries.py

```diff
@@ -58,7 +57,7 @@ class TreeManager(TreeManager_, BaseManager.from_queryset(TreeQuerySet)):
     _with_tree_fields = True
     use_in_migrations = True
 
-    @cached_property
+    @limited_lru_cache(max_size=16)
```

Comment:
`@lru_cache` is not a drop in replacement for `@cached_property` (method versus property). 🤦 Reworking this, stand by.

Reply:
cached_property should be fine to leave since it's only caching instance attributes?

Reply:
Because you should never have an instance shared by multiple threads?

Reply:
Issue isn't with sharing by multiple threads, it's the long life of the cached_property and the fact that signals only clear the cache for the process that triggered the signal, so other workers may retain stale cache data. I think you're right that switching to using a shared Django cache is the preferred approach here, working on that now.